### PR TITLE
Pretty error messages

### PR DESCRIPTION
**Pull request by jacobwindsor**
- Adds an error module
- Redirects all global JS errors to general error page
- Redirects 404s to 404 page
- Redirects 500s to 500 page
- Redirects all other HTTP failures to general error page

### CODE REVIEW

**Review by MightyAx — DISMISSED** [review]
Minor edits to open and closing tags, but if there's a good reason it's fine.

File: src/app/components/global-header/global-header.component.html [hunk 1]
```diff
@@ -1,7 +1,17 @@
...
+<section
+  class="vf-hero | vf-u-fullbleed vf-hero-hca"
+  style="background-color: #3355aa"
+>
   <div class="vf-hero__content | vf-box | vf-stack vf-stack--400">
-    <h2 class="vf-hero__heading"><a class="vf-hero__heading_link" routerLink="/">Human Cell Atlas Project Catalogue</a></h2>
-    <p class="vf-hero__subheading">A comprehensive list of cellular resolution datasets, selected and curated by the HCA DCP data wrangling team</p>
+    <h2 class="vf-hero__heading">
+      <a class="vf-hero__heading_link" routerLink="/"
+        >Human Cell Atlas Project Catalogue</a
```
[the remark this answers is not in the document]
Comment (jacobwindsor) [reply src/app/components/global-header/global-header.component.html:8]: This is the formatting done by prettier. I agree with the second one - this looks weird. You can't change the formatting done by prettier and in general I like the style. Maybe we should investigate using ESLint instead but for another ticket?

**Comment by sandykadam** [comment]
@jacobwindsor Since you created custom error pages, you might want to add in conf file as well, so Nginx will redirect to respective error for e.g

```
error_page   500 502 503 504  /humancellatlas/project-catalogue/error;
```

https://github.com/ebi-ait/projects-index/blob/master/docker-assets/default.conf#L12

https://github.com/ebi-ait/projects-index/blob/master/docker-assets/default.conf#L16

Check in local first with Docker build.

**Comment by jacobwindsor** [comment]
@sandykadam I have done that now! thanks for the suggestion

File: src/app/components/global-header/global-header.component.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (amnonkhen) [reply src/app/components/global-header/global-header.component.html:8]: Maybe the [jsx brackets](https://prettier.io/docs/en/options.html#jsx-brackets) could help?

File: docker-assets/default.conf [hunk 2]
```diff
@@ -9,14 +9,10 @@ server {
         try_files $uri $uri/ /humancellatlas/project-catalogue/index.html =404;
     }
 
-    error_page  404              /not-found.html;
+    error_page  404              /humancellatlas/project-catalogue/error/404;
+    error_page  500              /humancellatlas/project-catalogue/error/500;
```
Comment (amnonkhen) [inline docker-assets/default.conf:13]: Why do we differentiate between 500 & 50x errors?

**Review by amnonkhen — APPROVED** [review]

**Review by amnonkhen — CHANGES_REQUESTED** [review]
Looks well.
I agree with @MightyAx's comments about closing bracket location.

**Review by ke4 — DISMISSED** [review]
I agree with @amnonkhen and @MightyAx 's findings (formatting issues), but other than those it looks fine to me 👍 💯

**Comment by jacobwindsor** [comment]
@amnonkhen I don't know where your comment about 500 pages has gone but it is because the VF [500 page](https://stable.visual-framework.dev//patterns/error-pages/index.html?q=error#500) has an error specifically for 500

**Review by jacobwindsor — COMMENTED** [review]
The issue with HTML formatting is being actively worked on in this [PR](https://github.com/prettier/prettier/issues/5377#issuecomment-678644985). So, I don't think we can do anything about this now

**Comment by amnonkhen** [comment]
@jacobwindsor Do you mean that prettier cannot currently put the bracket at the end of the previous line? I will approve for now, but we should sort this out soon. Another code formatter might be a good idea here.

**Review by amnonkhen — DISMISSED** [review]

**Review by ke4 — APPROVED** [review]

**Review by amnonkhen — APPROVED** [review]

**Comment by jacobwindsor** [comment]
Merging now but there is still work to be done for this in [this card](https://app.zenhub.com/workspaces/dcp-ingest-product-development-5f71ca62a3cb47326bdc1b5c/issues/ebi-ait/dcp-ingest-central/334)

**Review by ke4 — APPROVED** [review]

**Review by amnonkhen — APPROVED** [review]

**Pull request merged**